File: prime/src/main/java/com/telenordigital/prime/storage/entities/PurchaseRequest.java
package com.telenordigital.prime.storage.entities;

import java.util.HashMap;
import java.util.Map;

public interface PurchaseRequest {
    String getSku();

    String getPaymentToken();

    String getMsisdn();

    long getMillisSinceEpoch();

    String getId();

    default Map<String, Object> asMap() {
        final Map<String, Object> result = new HashMap<>();
        result.put("msisdn", getMsisdn());
        result.put("sku", getSku());
        result.put("paymentToken", getPaymentToken());
        return result;
    }
}

File: prime/src/main/kotlin/org/ostelco/prime/storage/entities/PurchaseRequest.kt
package org.ostelco.prime.storage.entities

import java.util.*

interface PurchaseRequest {
    val sku: String

    val paymentToken: String

    val msisdn: String

    val millisSinceEpoch: Long

    val id: String

    fun asMap(): Map<String, Any> {
        val result = HashMap<String, Any>()
        result["msisdn"] = msisdn
        result["sku"] = sku
        result["paymentToken"] = paymentToken
        return result
    }
}
